docs/sql-distributed-sql-engine-spark-sql-cli.md

```diff
@@ -0,0 +1,185 @@
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+* Table of contents
+{:toc}
+
+
+The Spark SQL CLI is a convenient tool to run the Hive metastore service in local mode and execute
+queries input from the command line. Note that the Spark SQL CLI cannot talk to the Thrift JDBC server.
+
+To start the Spark SQL CLI, run the following in the Spark directory:
+
+    ./bin/spark-sql
+
+Configuration of Hive is done by placing your `hive-site.xml`, `core-site.xml` and `hdfs-site.xml` files in `conf/`.
+
+## Spark SQL Command Line Options
+
+You may run `./bin/spark-sql --help` for a complete list of all available options.
+
+    CLI options:
+     -d,--define <key=value>          Variable substitution to apply to Hive
+                                      commands. e.g. -d A=B or --define A=B
+        --database <databasename>     Specify the database to use
+     -e <quoted-query-string>         SQL from command line
+     -f <filename>                    SQL from files
+     -H,--help                        Print help information
+        --hiveconf <property=value>   Use value for given property
+        --hivevar <key=value>         Variable substitution to apply to Hive
+                                      commands. e.g. --hivevar A=B
+     -i <filename>                    Initialization SQL file
+     -S,--silent                      Silent mode in interactive shell
+     -v,--verbose                     Verbose mode (echo executed SQL to the
+                                      console)
+
+## The hiverc File
+
+The Spark SQL CLI when invoked without the `-i` option will attempt to load `$HIVE_HOME/bin/.hiverc` and `$HOME/.hiverc` as initialization files.
+
+## Supported comment types
+
+<table class="table">
+<tr><th>Comment</th><th>Example</th></tr>
+<tr>
+  <td>simple comment</td>
+  <td>
+  <code>
+      -- This is a simple comment.
+      <br>
+      SELECT 1;
+  </code>
+  </td>
+</tr>
+<tr>
+  <td>bracketed comment</td>
+  <td>
+    <code>
+        /* This is a bracketed comment. */
+        <br>
+        SELECT 1;
+    </code>
+  </td>
+</tr>
+<tr>
+  <td>nested bracketed comment</td>
+  <td>
+    <code>
+        /*  This is a /* nested bracketed comment*/ .*/
+        <br>
+        SELECT 1;
+    </code>
+  </td>
+</tr>
+</table>
+
+## Spark SQL CLI Interactive Shell Commands
+
+When `./bin/spark-sql` is run without either the `-e` or `-f` option, it enters interactive shell mode.
+Use `;` (semicolon) to terminate commands. Notice:
+1. CLI use `;` to terminate commands only when it's at the end of line and it's not escaped by `\\;`.
+2. `;` is the only way to terminate commands, if user type `SELECT 1` and press enter, console will just wait for input.
+3. If user type multiple commands in one line like `SELECT 1; SELECT 2;`, commands `SELECT 1` and `SELECT 2` will be executed separatly.
+4. If `;` in a simple comment 
```

Comment:
If `;` appears in a simple comment, as in:

Reply:
Done